lib/validate.ts

```diff
@@ -21,13 +21,23 @@ export const ajv = new Ajv({
   verbose: isDebugEnabled,
 });
 
+type ErrorIsModuleNotFoundException = (
```

Comment:
Weirdly enough we need to provide that, otherwise it'll pop up this error:
```
Assertions require every name in the call target to be declared with an explicit type annotation.ts(2775)
errors.ts(10, 14): 'assertIsError' needs an explicit type annotation.
```

Reply:
thats weird, is it ts or eslint error?

Reply:
It is a typescript error, according to [this issue](https://github.com/microsoft/TypeScript/issues/41047) seems it seems to be like this by design

Reply:
It can be defined at once. But the type must be specified.
```typescript
type AssertIsErrorFn = (e: unknown) => asserts e is Error;
const assertIsError: AssertIsErrorFn = (e) => if(!(e instanceof Error) throw e);
```
Is the same as
```typescript
const assertIsError: (e: unknown) => asserts e is Error; = (e) => if(!(e instanceof Error) throw e);
```

- Assertion functions `cannot` have their type inferred from the assignment.

Reply:
Good catch! @DDDKnightmare